delta_shooter/models.py

```diff
@@ -0,0 +1,277 @@
+        self.size = size
+        self.rect = pygame.Rect(0, 0, size[0], size[1])
+
+    def get_size(self) -> Tuple[int, int]:
+        return self.size[0], self.size[1]
+
+    def get_rect(self) -> pygame.Rect:
+        return self.rect
+
+
+class DummySprite(ABC):
+    @abstractmethod
+    def get_width(self) -> int:
+        pass
+
+
+class DummyShip(DummySprite):
+    def get_width(self) -> int:
+        return 40
+
+
+class DummyBullet(DummySprite):
+    def get_width(self) -> int:
+        return 10
+
+
+class DummySound:
+    def play(self) -> None:
+        pass
+
+
+Orientation = Literal["horizontal", "vertical"]
+
+
+class GameObject:
+    def __init__(
+        self,
+        starting_position: Union[Tuple[int, int], Vector2],
+        sprite: Union[Surface, DummySprite],
+        velocity: Union[int, Vector2],
+    ) -> None:
+        self.set_position(starting_position)
+        self.sprite = sprite
+        self.radius = int(sprite.get_width() / 2)
+        self.velocity = Vector2(velocity)
+        self.face_up()
+
+    def set_position(self, position: Union[Tuple[int, int], Vector2]) -> None:
+        self.position = Vector2(position)
+
+    def set_orientation(self, orientation: Vector2) -> None:
+        self.direction = Vector2(orientation)
+
+    def face_up(self) -> None:
+        self.set_orientation(UP)
+
+    @property
+    def angle(self) -> int:
+        return round(self.direction.angle_to(UP))
+
+    def draw(self, surface: pygame.surface.Surface) -> None:
+        assert isinstance(self.sprite, pygame.Surface)
+        blit_position = self.position - Vector2(self.radius)
+        surface.blit(self.sprite, blit_position)
+
+    def move(self, surface: Union[pygame.surface.Surface, DummyScreen]) -> None:
+        self.position = edge_barriers(self.position + self.velocity, self.radius, surface)
+
+    def collides_with(self, other_obj: "GameObject") -> bool:
+        """Fudge factor stops bullets skipping over objects."""
+
+        fudge_factor = 1.5
+        distance = self.position.distance_to(other_obj.position)
+        return distance < (self.radius * fudge_factor) + (other_obj.radius * fudge_factor)
+
+
+class Spaceship(GameObject):
+    ANGLE_TURN = 15
+    ACCELERATION = 0.1
+    BULLET_SPEED = 60
+    NUM_BULLETS = 2  # Limit the number on the screen at one time
+
+    def __init__(
+        self,
+        starting_position: Tuple[int, int],
+        starting_orientation: Vector2,
+        player: int,
+        graphical: bool = True,
+    ) -> None:
+
+        self.starting_position = starting_position
+        self.starting_orientation = starting_orientation
+        self.graphical = graphical
+        self.player = player
+        self.dead = False
+        self.name = "spaceship"
+
+        if self.graphical:
+            super().__init__(
+                starting_position, load_sprite(f"spaceship_player{player}"), Vector2(0)
+            )
+            self.laser_sound = load_sound("laser")
```

Comment:
```suggestion
            try:
                self.laser_sound = load_sound("laser")
            except pygame.error:
                self.laser_sound = DummySound()
```

Otherwise I get a:
```
Traceback (most recent call last):
  File "/home/henry/delta_repos/shooter/delta_shooter/main.py", line 74, in <module>
    play_shooter(
  File "/home/henry/delta_repos/shooter/delta_shooter/game_mechanics.py", line 50, in play_shooter
    game = ShooterEnv(
  File "/home/henry/delta_repos/shooter/delta_shooter/game_mechanics.py", line 109, in __init__
    self.reset()
  File "/home/henry/delta_repos/shooter/delta_shooter/game_mechanics.py", line 124, in reset
    self.player1 = Spaceship(
  File "/home/henry/delta_repos/shooter/delta_shooter/models.py", line 123, in __init__
    self.laser_sound = load_sound("laser")
  File "/home/henry/delta_repos/shooter/delta_shooter/shooter_utils.py", line 28, in load_sound
    return Sound(str(path))
pygame.error: mixer not initialized

Process finished with exit code 1
```

Reply:
old man yells at linux